cloudant-http/src/main/java/com/cloudant/http/HttpConnection.java

```diff
@@ -274,18 +275,19 @@ public HttpConnection execute() throws IOException {
                     connection.setChunkedStreamingMode(1024);
                 }
 
-                int bufSize = 1024;
-                int nRead = 0;
-                byte[] buf = new byte[bufSize];
                 InputStream is = input.getInputStream();
                 OutputStream os = connection.getOutputStream();
 
-                while ((nRead = is.read(buf)) >= 0) {
-                    os.write(buf, 0, nRead);
+                try {
+                    try {
+                        IOUtils.copy(is, os);
+                        os.flush();
+                    } finally {
```

Comment:
I've removed one of them in 8891940.
As discussed I think one needs to remain in case the `flush` throws an exception.